addons/point_of_sale/models/pos_order.py

```diff
@@ -1400,6 +1400,17 @@ def write(self, values):
                     del pl[2]['server_id']
         return super().write(values)
 
+    def get_existing_lots(self, company_id, product_id):
```

Comment:
Do we really need company_id? Can't we get it with self.company_id?

Reply:
I use this method to get existing lots from the database here: https://github.com/odoo/odoo/pull/127533/files#diff-24c238625928c3ef5f7c6957ed86c4feafa8beffefe2a15c502e2ba57b1ab26fR209
Self is therefore empty.